admin-dev/themes/new-theme/js/components/entity-search-input.js

```diff
@@ -82,6 +86,103 @@ export default class EntitySearchInput {
     });
   }
 
+  /**
+   * @param {string} optionName
+   * @param {*} value
+   */
+  setOption(optionName, value) {
+    this.options[optionName] = value;
+  }
+
+  /**
+   * @param {Object} options
+   */
+  buildOptions(options) {
+    const inputOptions = options || {};
+    this.initOption('mappingValue', inputOptions, 'id');
+    this.initOption('mappingDisplay', inputOptions, 'name');
+    this.initOption('mappingImage', inputOptions, 'image');
+
+    this.initOption('prototypeTemplate', inputOptions);
+    this.initOption('prototypeName', inputOptions, '__name__');
+    this.initOption('prototypeImage', inputOptions, '__image__');
+    this.initOption('prototypeValue', inputOptions, '__value__');
+    this.initOption('prototypeDisplay', inputOptions, '__display__');
+
+    this.initOption('allowDelete', inputOptions, true);
+    this.initOption('dataLimit', inputOptions, 0);
+    this.initOption('remoteUrl', inputOptions);
+
+    this.initOption('removeModalTitle', inputOptions);
+    this.initOption('removeModalMessage', inputOptions);
+    this.initOption('removeModalApply', inputOptions);
+    this.initOption('removeModalCancel', inputOptions);
+
+    // Most of the previous config are configurable via the EntitySearchInputForm options, the following ones are only
+    // overridable via js config (as long as you use the default template)
+    this.initOption('searchInputSelector', inputOptions, '.entity-search-input');
+    this.initOption('listSelector', inputOptions, '.entities-list');
+    this.initOption('removeModalId', inputOptions, 'modal-confirm-remove-entity');
+    this.initOption('confirmButtonClass', inputOptions, 'btn-danger');
+    this.initOption('queryWildcard', inputOptions, '__QUERY__');
+    this.initOption('entityItemSelector', inputOptions, 'li.entity-item');
+    this.initOption('entityDeleteSelector', inputOptions, '.entity-item-delete');
```

Comment:
Ping @Progi1984, please moove every selectors to a selectors map 😗

Reply:
@jolelievre do we really need all these options methods? You could set default values inside this.options, spread the inputOptions inside this.options then loop and checks

Reply:
As discussed together, let's add this to a default object, for those who doesn't have default value, set these as undefined

This will provide a better interface, and will be strictly typed with typescript

And then loop over, using initOption :)

Reply:
Mmmm, I'm not sure it's relevant in that case since they are default fallback values The selector map makes sense when you need to override these default ones in which case new custom selectors would be in a map

But here I don't know if t's worth it, unless you think I should introduce a dedicated selector map for this component alone like `EntitySearchInputMap.js` ?

Reply:
> Mmmm, I'm not sure it's relevant in that case since they are default fallback values The selector map makes sense when you need to override these default ones in which case new custom selectors would be in a map
> 
> But here I don't know if t's worth it, unless you think I should introduce a dedicated selector map for this component alone like `EntitySearchInputMap.js` ?

Use the components/selectors-map, add a new key entitySearchInput

Selectors map is not only designed for override, but it permit to have only one path where you use constants, so if someone modify this class, he can easily detect the file where it's used 

But we're using it a bit wrongly and the QA team doesn't take benefits on it for the moment.. I know it's frustrating (@Progi1984 makes my head exploding on this too)

Reply:
It's still kind of useful to share selectors between components I agree

I ended up creating a dedicated `admin-dev/themes/new-theme/js/components/entity-search-input-map.js` for now, I didn't see your message before and the `components/selectors-map` is not merged on `develop` yet

Do you prefer me to create this file in my PR as well? And we will fix conflicts later since it will have been created in two different PRs? (no problem for me to do that)